### ISSUE TITLE
SELECT COLUMNS() support lambda filter

### ISSUE THREAD

**Issue by danthegoodman1** [ClickHouse/ClickHouse#54461]
> (you don't have to strictly follow this form)

**Use case**

Dynamic column selection based on lambda function results (boolean match)

**Describe the solution you'd like**

Example of selecting columns where the sum of the chars are greater than 5 (terrible example)

```
columns(x -> arraySum(y -> ascii(y), splitByString('', x)) > 5)
```

**Comment by den-crane** [comment]
https://youtu.be/zhrOYQpgvkk?t=765

**Comment by den-crane** [comment]
I guess it was implemented with a new analyzer

**Comment by danthegoodman1** [comment]
@den-crane this seems to be apply to the results, but not actually applying lambdas to which columns will be filtered? It's gone over quite quickly and not explictly mentioned in that section as far as I can tell...

**Comment by den-crane** [comment]
Not sure I understand you https://fiddle.clickhouse.com/a157f08e-f9af-4a3a-9256-02a03101e1be

**Comment by danthegoodman1** [comment]
Yes I mean the dynamic number of columns, similar to https://duckdb.org/docs/sql/query_syntax/select.html#syntax with the lambda functions

**Comment by den-crane** [comment]
this is not possible with Map, and contradicts SQL principals, in Clickhouse number and types of columns in the resultset are known before a query execution.
```
SELECT UNNEST({'a': 42, 'b': 84});
┌───────┬───────┐
│   a   │   b   │
│ int32 │ int32 │
├───────┼───────┤
│    42 │    84 │
└───────┴───────┘
```

this is possible 

```
create table A(a String, col1 String, col2 String)
Engine = Memory as select '333', '111', '222';

select columns('col\d+') 
from A
format Pretty;
+------+------+
| col1 | col2 |
+------+------+
| 111  | 222  |
+------+------+
```
